components/molecule/autosuggestField/src/index.js

```diff
@@ -17,57 +17,53 @@ const getState = ({successText, errorState, alertText}) => {
   if (alertText) return MoleculeAutosuggestStates.ALERT
 }
 
-class MoleculeAutosuggestField extends Component {
-  refAutosuggest = React.createRef()
-
-  handleClick = () => {
-    const {current: domAutosuggest} = this.refAutosuggest
+function MoleculeAutosuggestField({
+  alertText,
+  errorText,
+  helpText,
+  id,
+  inline,
+  label,
+  onChange,
+  successText,
+  useContrastLabel,
+  children,
+  ...restProps
+}) {
+  const refAutosuggest = useRef()
+
+  const handleClick = () => {
```

Comment:
better memoized

```
const handleClick = useCallback( () => {
  const {current: domAutosuggest} = refAutosuggest
  if (domAutosuggest) domAutosuggest.focus()
}, [refAutosuggest])
```

Reply:
I understand that you say it for performance, but I think that in this case it is not necessary.

The first is that the optimization can be more expensive than the benefit, for this case I think that it is more advantageous to execute the same code in each render, than to delegate the cost to React to check the change of the dependencies.

And the other reason is that in this case we are using a variable that has been created with `useRef` and does not change between different renders.

https://kentcdodds.com/blog/usememo-and-usecallback